lightclient/src/background.rs

```diff
@@ -100,87 +227,67 @@ struct PendingSubscription {
     /// Send the method response ID back to the user.
     ///
     /// It contains the subscription ID if successful, or an JSON RPC error object.
-    sub_id_sender: oneshot::Sender<MethodResponse>,
-    /// The subscription state that is added to the `subscriptions` map only
-    /// if the subscription ID is successfully sent back to the user.
-    subscription_state: ActiveSubscription,
-}
-
-impl PendingSubscription {
-    /// Transforms the pending subscription into an active subscription.
-    fn into_parts(self) -> (oneshot::Sender<MethodResponse>, ActiveSubscription) {
-        (self.sub_id_sender, self.subscription_state)
-    }
+    response_sender: oneshot::Sender<SubscriptionResponse>,
+    /// The unsubscribe method to call when the user drops the receiver
+    /// part of the channel.
+    unsubscribe_method: String,
 }
 
 /// The state of the subscription.
 struct ActiveSubscription {
     /// Channel to send the subscription notifications back to frontend.
-    sender: mpsc::UnboundedSender<Box<RawValue>>,
+    notification_sender: mpsc::UnboundedSender<Result<Box<RawValue>, JsonRpcError>>,
```

Comment:
I tweaked subscriptions a bit to:

a) return any subscription notification errors that come back (previously these were being ignored I think) and
b) send back the ID and repsonse channel rather than one channel for ID and one for responses, just to simplify usage a little.